File: M3-Programacion/M3/Practica11.java
import java.io.*;

/**
 * Exercici 1: Cerca un caràcter
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix un text breu: ");
		String text = r.readLine();
		System.out.print("Introdueix una lletra per buscarla al text: ");
		char lletra = r.readLine().charAt(0);
		int num = text.indexOf(lletra);
		if (num == -1) {
			System.out.println("El caràcter '" + lletra + "' no apareix a \"" + text + "\"");
		} else {
			System.out.println("El caràcter '" + lletra + "' apareix a \"" + text + "\"");
		}
	}
}*/

/**
 * Exercici 2: Nom en majúscules
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix un text breu: ");
		String text = r.readLine();
		text = text.toUpperCase();
		System.out.print(text);
	}
}*/

/**
 * Exercici 3: El nombre de A's
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix el teu nom: ");
		String text = r.readLine();
		int num = 0;
		for (int i = 0; i < text.length(); i++) {
			if (text.charAt(i) == 'a') {
				num += 1;
			}
		}
		System.out.print("El nom conté " + num + " 'a's");
	}
}*/

/**
 * Exercici 4: Anàlisi del nom
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix el teu nom: ");
		String nom = r.readLine().toLowerCase();
		int length = nom.length();
		int vocals = 0;
		for (int i = 0; i < length; i++) {
			if (nom.charAt(i) == 'a' || nom.charAt(i) == 'o' || nom.charAt(i) == 'i' || nom.charAt(i) == 'e' || nom.charAt(i) == 'u') {
				vocals++;
			}
		}
		System.out.println(length + " càracters i " + vocals + " vocals");
	}
}*/

/**
 * Exercici 5: Inicials en majúscules
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix el teu nom: ");
		String nom = r.readLine().toLowerCase();
		int length = nom.length();
		int vocals = 0;
		for (int i = 0; i < length; i++) {
			if (nom.charAt(i) == 'a' || nom.charAt(i) == 'o' || nom.charAt(i) == 'i' || nom.charAt(i) == 'e' || nom.charAt(i) == 'u') {
				vocals++;
			}
		}
		System.out.println(length + " càracters i " + vocals + " vocals");
	}
}*/

/**
 * Exercici 6: Codi Morse
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix el text per traduir a morde: ");
		String text = r.readLine().toLowerCase();
		String morse = "morse: ";
		char l[] = {'a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z','1','2','3','4','5','6','7','8','9','0'};
		String m[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--..",".----","..---","...--","....-",".....","-....","--...","---..","----.","-----"};
		for (int i = 0; i < text.length(); i++) {
			for (int n = 0; n < l.length; n++) {
				if (text.charAt(i) == l[n]) {
					morse += m[n] + " ";
					break;
				}
			}
		}
		System.out.println(morse);
	}
}*/

/**
 * Exercici 7: Figures amb strings
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introduce una paraula: ");
		String paraula = r.readLine();
		String text;
		int l = paraula.length();
		for (int i = l - 1; i > 0; i--) {
			text = "";
			for (int n = i; n < l; n++) {
				text += paraula.charAt(n);
			}
			System.out.println(text);
		}
		for (int i = l; i > 0; i--) {
			text = "";
			for (int n = 0; n < i; n++) {
				text += paraula.charAt(n);
			}
			System.out.println(text);
		}
	}
}*/

/**
 * Exercici 8: La piràmide
 **/
/*public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("Introdueix una paraula: ");
		String paraula = r.readLine();
		int l = paraula.length();
		for (int i = 1; i < l + 1; i++) {
			String lef = "";
			String rig = "";
			for (int n =0; n < i; n++) {
				if (n + 1 == i) {
					lef += paraula.charAt(n);
					rig = paraula.charAt(n) + rig;
				}
				else{
					lef += paraula.charAt(n) + ",";
					rig = "," + paraula.charAt(n) + rig;
				}
			}
				
			for (int n = 0; n < l - i; n++) {
				lef = ". " + lef;
				rig += " .";
			}
			System.out.println(lef + "=" + rig);
		}
	}
}*/

/**
 * Exercici 9: Cerca d'un string
 **/
public class Practica11 {
	
	public static void main (String args[]) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("String: ");
		String string = r.readLine();
		System.out.print("Substring: ");
		String substring = r.readLine();
		if (string.equals("")) {
			System.out.println("L'string no conte \"" + substring + '"');
		} else if (substring.equals("")) {
			System.out.println("Esta inclos a cualsevol string no buit");
		} else {
			if (string.indexOf(substring) != -1) {
				System.out.println('"' + substring + "\" es troba a l'string \"" + string + '"');
			} else {
				System.out.println('"' + substring + "\" no i es a l'string \"" + string + '"');
			}
		}
	}
}

File: M3-Programacion/M3/Practica11.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


# Exercici 1: Cerca un caràcter
"""
text = raw_input("Introdueix un text breu: ")
lletra = raw_input("Introdueix una lletra per buscarla al text: ")
if lletra in text: 
	print "El caràcter '" + lletra + "' apareix a \"" + text + '"'
else:
	print "El caràcter '" + lletra + "' no apareix a \"" + text + '"'
"""

# Exercici 2: Nom en majúscules
"""
text = raw_input("Escriu el teu nom: ")
print text.upper()
"""

# Exercici 3: El nombre de A's
"""
text = raw_input("Escriu el teu nom: ")
num = 0
for i in text:
	if i == 'a':
		num +=1
print "El nom conté", num, "'a's"
"""

# Exercici 4: Anàlisi del nom
"""
nom = raw_input("Introdueix el teu nom: ")
length = len(nom)
vocals = 0
for i in nom:
	if i == 'a' or i == 'e' or i == 'i' or i == 'o' or i == 'u':
		vocals += 1
print length, "càracters i",vocals,"vocals"
"""

# Exercici 5: Inicials en majúscules
"""
nom = raw_input("Introdueix el teu nom complet: ")
nomupper = nom[0].upper()
for i in range(1, len(nom), 1):
    if nom[i - 1] == " ":
        nomupper += nom[i].upper()
    else:
        nomupper += nom[i].lower()
print nomupper
"""

# Exercici 6: Codi Morse
"""
text = raw_input("Introdueix el text per traduir a morde: ")
text = text.lower()
morse = "morse: "
l = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z','1','2','3','4','5','6','7','8','9','0']
m = ['.-', '-...', '-.-.', '-..', '.', '..-.', '--.', '....', '..', '.---', '-.-', '.-..', '--', '-.', '---', '.--.', '--.-', '.-.', '...', '-', '..-', '...-', '.--', '-..-', '-.--', '--..', '.----', '..---', '...--', '....-', '.....', '-....', '--...', '---..', '----.', '-----']
for i in range(len(text)):
    for n in range(len(l)):
        if text[i] == l[n]:
            morse += m[n] + " "
            break
print morse
"""

# Exercici 7: Figures amb strings
"""
paraula = raw_input("Introduce una paraula: ")
l = len(paraula)
for i in range(l - 1, 0, -1):
    print paraula[i:]
for i in range(l, 0, -1):
    print paraula[:i]
"""


# Exercici 8: La piràmide
"""
paraula = raw_input("Introdueix una paraula: ")
l = len(paraula)
for i in range(1, l + 1, 1):
    lef = ""
    rig = ""
    for n in range(i):
        if n + 1 == i:
            lef += paraula[n]
            rig = paraula[n] + rig
        else:
            lef += paraula[n] + ","
            rig = "," + paraula[n] + rig
        
    for n in range(l - i):
        lef = ". " + lef
        rig += " ."
    print lef + "=" + rig
"""


# Exercici 9: Cerca d'un string

string = raw_input("String: ")
substring = raw_input("Substring: ")
if string == "":
    print "L'string no conte " + substring
elif substring == "":
    print "Esta inclos a cualsevol string no buit"
else:
    if substring in string:
        print substring + " es troba a l'string \"" + string + '"'
    else:
        print substring + " no i es a l'string \"" + string + '"'

